rust/opendp/src/trans/resize/mod.rs

```diff
@@ -1,20 +1,23 @@
-use crate::core::{Transformation, Function, StabilityRelation};
+use crate::core::{Transformation, Function, StabilityRelation, Domain};
 use crate::error::Fallible;
 use crate::dist::{SymmetricDistance, IntDistance};
-use crate::dom::{VectorDomain, AllDomain, SizedDomain};
+use crate::dom::{VectorDomain, SizedDomain};
 use std::cmp::Ordering;
 use crate::traits::CheckNull;
 
-pub fn make_resize_constant<TA: 'static + Clone + CheckNull>(
-    constant: TA, length: usize
-) -> Fallible<Transformation<VectorDomain<AllDomain<TA>>, SizedDomain<VectorDomain<AllDomain<TA>>>, SymmetricDistance, SymmetricDistance>> {
+pub fn make_resize_constant<DA>(
+    atom_domain: DA,
+    constant: DA::Carrier, length: usize
+) -> Fallible<Transformation<VectorDomain<DA>, SizedDomain<VectorDomain<DA>>, SymmetricDistance, SymmetricDistance>>
+    where DA: 'static + Clone + Domain,
+          DA::Carrier: 'static + Clone + CheckNull {
+    if !atom_domain.member(&constant)? { return fallible!(MakeTransformation, "constant must be a member of DA")}
```

Comment:
Calling `member()` here feels a little funky. I guess this is a bit of a gap in the framework, and the way we've modeled Domain types. I can't think of a better way to handle this, but I think it's worth a discussion with Salil.